flink-ml-lib/src/main/java/org/apache/flink/ml/common/BatchOperator.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.flink.ml.common;
+
+import org.apache.flink.ml.param.Param;
+import org.apache.flink.table.api.Table;
+
+import java.util.Arrays;
+import java.util.Map;
+
+/**
+ * Base class of offline learning algorithm operators.
+ *
+ * <p>This class extends {@link BaseAlgoImpl} to support data transmission between BatchOperators.
+ */
+public abstract class BatchOperator<T extends BatchOperator<T>> extends BaseAlgoImpl<T> {
```

Comment:
Hmm.. I thought the plan is to postphone the addition and the discussion of `BatchOperator` and `BaseAlgoImpl`. Could we start simple for now and remove them from this PR?

Reply:
I will remove them later.

Reply:
done